File: sdk/python/helix_seller/merchant.py
from typing import List, Optional
from .client import HelixClient


class MerchantService:
    def __init__(self, client: HelixClient):
        self.client = client

    def list(self) -> List[dict]:
        result = self.client.get("/api/v1/merchants")
        return result.get("merchants", [])

    def get(self, merchant_id: str) -> dict:
        return self.client.get(f"/api/v1/merchants/{merchant_id}")

    def create(
        self,
        legal_name: str,
        email: str,
        country: str,
        currency: str,
        trade_name: Optional[str] = None,
        phone: Optional[str] = None,
    ) -> dict:
        return self.client.post(
            "/api/v1/merchants",
            {
                "legal_name": legal_name,
                "email": email,
                "country": country,
                "currency": currency,
                **({"trade_name": trade_name} if trade_name else {}),
                **({"phone": phone} if phone else {}),
            },
        )

    def update(self, merchant_id: str, **kwargs) -> dict:
        return self.client.put(f"/api/v1/merchants/{merchant_id}", kwargs)

File: cmd/cli/merchant.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"text/tabwriter"

	"github.com/spf13/cobra"
)

func merchantCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "merchant",
		Short: "Manage merchants",
	}

	cmd.AddCommand(
		merchantListCmd(),
		merchantGetCmd(),
		merchantCreateCmd(),
	)

	return cmd
}

func merchantListCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "list",
		Short: "List all merchants",
		RunE: func(cmd *cobra.Command, args []string) error {
			data, err := apiGet("/api/v1/merchants")
			if err != nil {
				return err
			}
			var result struct {
				Merchants []map[string]interface{} `json:"merchants"`
			}
			json.Unmarshal(data, &result)

			w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
			fmt.Fprintln(w, "ID\tNAME\tSTATUS\tCOUNTRY")
			for _, m := range result.Merchants {
				fmt.Fprintf(w, "%s\t%s\t%s\t%s\n",
					m["id"], m["legal_name"], m["status"], m["country"])
			}
			w.Flush()
			return nil
		},
	}
}

func merchantGetCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "get [id]",
		Short: "Get merchant details",
		Args:  cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			data, err := apiGet("/api/v1/merchants/" + args[0])
			if err != nil {
				return err
			}
			var m map[string]interface{}
			json.Unmarshal(data, &m)
			w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
			for k, v := range m {
				fmt.Fprintf(w, "%s:\t%v\n", k, v)
			}
			w.Flush()
			return nil
		},
	}
}

func merchantCreateCmd() *cobra.Command {
	var legalName, email, country, currency string

	cmd := &cobra.Command{
		Use:   "create",
		Short: "Create a new merchant",
		RunE: func(cmd *cobra.Command, args []string) error {
			body := map[string]string{
				"legal_name": legalName,
				"email":      email,
				"country":    country,
				"currency":   currency,
			}
			data, err := apiPost("/api/v1/merchants", body)
			if err != nil {
				return err
			}
			var m map[string]interface{}
			json.Unmarshal(data, &m)
			fmt.Printf("Merchant created: %s\n", m["id"])
			return nil
		},
	}

	cmd.Flags().StringVar(&legalName, "name", "", "Legal name")
	cmd.Flags().StringVar(&email, "email", "", "Email")
	cmd.Flags().StringVar(&country, "country", "", "Country code")
	cmd.Flags().StringVar(&currency, "currency", "USD", "Currency")
	cmd.MarkFlagRequired("name")
	cmd.MarkFlagRequired("email")
	cmd.MarkFlagRequired("country")

	return cmd
}
